File: app/src/main/java/com/thecoolguy/rumaan/fileio/ui/NoNetworkDialogFragment.java
package com.thecoolguy.rumaan.fileio.ui;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;

import com.thecoolguy.rumaan.fileio.R;


public class NoNetworkDialogFragment extends DialogFragment implements View.OnClickListener {
    private DialogClickListener mListener;

    public NoNetworkDialogFragment() {
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        // Verify that the host activity implements the callback interface
        try {
            // Instantiate the NoticeDialogListener so we can send events to the host
            mListener = (DialogClickListener) getActivity();
        } catch (ClassCastException e) {
            // The activity doesn't implement the interface, throw exception
            throw new ClassCastException(getActivity().toString()
                    + " must implement NoticeDialogListener");
        }
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        LayoutInflater inflater = getActivity().getLayoutInflater();
        View view = inflater.inflate(R.layout.dialog_no_network_error, null);

        Button okBtn = view.findViewById(R.id.btn_ok);
        okBtn.setOnClickListener(this);

        builder.setView(view);

        return builder.create();
    }

    @Override
    public void onClick(View view) {
        mListener.onDialogPositiveClick(getDialog());
    }

}

File: app/src/main/java/com/thecoolguy/rumaan/fileio/ui/fragments/NoNetworkDialogFragment.kt
package com.thecoolguy.rumaan.fileio.ui.fragments

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import androidx.appcompat.app.AlertDialog
import android.view.View
import android.widget.Button
import com.thecoolguy.rumaan.fileio.R
import com.thecoolguy.rumaan.fileio.listeners.DialogClickListener

class NoNetworkDialogFragment : androidx.fragment.app.DialogFragment(), View.OnClickListener {

  private var mListener: DialogClickListener? = null

  override fun onAttach(context: Context?) {
    super.onAttach(context)
    try {
      mListener = activity as DialogClickListener?
    } catch (e: ClassCastException) {
      throw ClassCastException(activity!!.toString() + " must implement NoticeDialogListener")
    }

  }

  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
    val builder = AlertDialog.Builder(activity!!)
    val inflater = activity!!.layoutInflater
    val view = inflater.inflate(R.layout.dialog_no_network_error, null)

    val okBtn = view.findViewById<Button>(R.id.btn_ok)
    okBtn.setOnClickListener(this)

    builder.setView(view)

    return builder.create()
  }

  override fun onClick(view: View) {
    mListener?.onDialogPositiveClick(dialog, this)
  }

}
